### PR TITLE
feat: Added share cart feature

### PR DESCRIPTION
**Pull request by jalajcodes**
https://max-store-app.herokuapp.com/

### CODE REVIEW

File: frontend/src/Components/CartPriceInfo.js [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+  const [shareText, setShareText] = useState("");
   const { total, discount, qty } = getPriceInfo(cartItems);
 
+  const handleShareCart = (items) => {
+    setShareText(null);
+    const itemIds = items.map((item) => item.product);
+    const encodedIds = Buffer.from(JSON.stringify(itemIds), "utf8").toString(
+      "base64"
+    );
+    const url = `${window.location.origin}/cart/shared/${encodedIds}`;
```
Comment (mikesch803) [inline frontend/src/Components/CartPriceInfo.js:15]: great feature

File: frontend/src/Pages/SharedCart.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      });
+
+      setCartData(data);
+    };
+
+    fetchCart();
+  }, [encodedIds]);
+
+  if (cartData.length === 0) {
+    return <NotFound />;
```
Comment (mikesch803) [inline frontend/src/Pages/SharedCart.js:31]: why you did this, not able to understand

**Review by mikesch803 — APPROVED** [review]
i commented your code where i m not understand why return not found page else your code looks good

File: frontend/src/Components/CartPriceInfo.js (hunk 1 above)
Comment (VLeads) [inline frontend/src/Components/CartPriceInfo.js:15]: Nice feature added !

**Review by VLeads — APPROVED** [review]
Nothing much to add . Great work

File: frontend/src/Pages/SharedCart.js (hunk 2 above)
Comment (jalajcodes) [reply frontend/src/Pages/SharedCart.js:31]: If no data found then show notfound component.

**Pull request merged**